os/net/netmgr/netstack_lwip.c

```diff
@@ -474,6 +475,34 @@ static int lwip_ns_getstats(int fd, struct netmon_sock **sock_info)
 }
 #endif
 
+static void lwip_ns_initlist(struct socketlist *list)
+{
+	for (int i = 0; i < CONFIG_NSOCKET_DESCRIPTORS; ++i) {
+		list->sl_sockets[i].sock = NULL;
+	}
+
+	/* Initialize the list access mutex */
+	sem_init(&list->sl_sem, 0, 1);
+}
+
+static void lwip_ns_releaselist(struct socketlist *list)
+{
+	int i;
+	int ret;
+	struct lwip_sock *psock;
+	for (i = 0; i < CONFIG_NSOCKET_DESCRIPTORS; ++i) {
+		if (list->sl_sockets[i].sock) {
+			psock = (struct lwip_sock *)list->sl_sockets[i].sock;
+			DEBUGASSERT(psock->conn);
+			ret = lwip_sock_close(psock);
+			if (ret) {
+				ndbg("socket could not close properly\n");
```

Comment:
There is no exit with error case. Is it ok to set it as NULL? It will be memory leak.

Reply:
corrected